xarray/core/common.py

```diff
@@ -539,24 +543,69 @@ def resample(self, freq, dim, how='mean', skipna=None, closed=None,
         """
         from .dataarray import DataArray
 
+        if dim is not None:
+            return self._resample_immediately(freq, dim, how, skipna, closed,
+                                              label, base, keep_attrs)
+
+        # More than one indexer is ambiguous, but we do in fact need one if
+        # "dim" was not provided, until the old API is fully deprecated
+        if len(indexer) != 1:
+            raise NotImplementedError(
```

Comment:
raising a `TypeError` instead of calling out to `self._resample_immediately()` will break backwards compatibility - there will be no way to use the old API. Do we want to make this break immediately?